packages/jest-haste-map/src/index.js

```diff
@@ -71,6 +74,8 @@ export type ModuleMap = HasteModuleMap;
 const NODE_MODULES = path.sep + 'node_modules' + path.sep;
 const VERSION = require('../package.json').version;
 
+let Watcher;
```

Comment:
That would allow us to spawn instances that use different watcher constructors (`WatchmanWatcher` or `NodeWatcher`). Is this what you have in mind?